src/movies.js

```diff
@@ -1,15 +1,110 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(array) {
+    //Create new Array and sort it
+    let newArray = array.slice().sort(function (a,b) {
+        return a.year - b.year || a.title.localeCompare(b.title);
+    })
+    return newArray;
+}
+
+/*const sorted = numbers.sort(function (a, b) {
+    return a - b;
+*/
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function howManyMovies(array) {
+    //Filter by Drama and Director
+    let number = array.filter(function (movie) {
+        return movie.genre.includes("Drama") && movie.director === "Steven Spielberg";
+    })
+    //Return length
+    return number.length;
+}
+
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(array) {
+    // Order alphabetically
+    let ordered = array.slice().sort(function (a,b) {
+        return a.title.localeCompare(b.title);
+    })
+    // Return only title of ordered
+    let onlyTitleOrdered = ordered.map(function(movie){
+        return movie.title;
+    })
+    //Return only 20 movies
+    return onlyTitleOrdered.slice(0,20);
+}
+
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(array) {
+    // Check if array empty
+    if (array == 0) return 0;
+    //Get only movie rates
+    let onlyRate = array.map(function(movie){
+        return movie.rate || 0;
```

Comment:
This works, though I'd personally use filter for this as it is exactly what filter excels at